### PR TITLE
release,make: add rule `make crate-publish` for publishing crates.io

### PR DESCRIPTION
**Pull request by flouthoc**
* Add new rule `make crate-publish` to automate publish at https://crates.io/crates/netavark
* Document steps for publishing to `crates.io`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -47,6 +49,24 @@ build: bin $(CARGO_TARGET_DIR)
 	cargo build $(release)
 	cp $(CARGO_TARGET_DIR)/$(profile)/netavark bin/netavark$(if $(debug),.debug,)
 
+.PHONY: crate-publish
+crate-publish:
+	@if [ "$(publish-version)" = "" ]; then\
+		echo "publish-version not set. Usage example 'make crate-publish publish-version=v0.1.2'";\
+		echo "Make sure repo already contains tag for the version";\
+		exit 1;\
+	fi
```
Comment (Luap99) [inline Makefile:58]: I would just call the variable `version`

File: Makefile [hunk 2]
```diff
@@ -47,6 +49,24 @@ build: bin $(CARGO_TARGET_DIR)
...
+	@if [ "$(REMOTE_ORIGIN_URL)" != "https://github.com/containers/netavark" ]; then\
+		echo "You are working with a fork please use actual repo while publishing. Please clone https://github.com/containers/netavark";\
+		echo "Make sure repo already contains tag for $(publish-version)";\
+		exit 1;\
+	fi
```
Comment (Luap99) [inline Makefile:63]: This looks wrong. Why am I forced to clone upstream as origin? My fork already has all the correct tags in it.

File: Makefile [hunk 3]
```diff
@@ -47,6 +49,24 @@ build: bin $(CARGO_TARGET_DIR)
...
+	git checkout tags/$(publish-version)
+	git branch
+	@echo "It is expected that you have already done 'cargo login' before running this command. If not command may fail later"
+	cargo publish --dry-run
+	cargo publish
+
```
Comment (Luap99) [inline Makefile:69]: you never switch back to the previous git state

**Review by Luap99 — COMMENTED** [review]
I don't really see why we should automatically switch git branches. I would expect the author to know that they have to be on the correct commit.
You already need to do a `make vendor` and `git tag` on this commit anyway so I don't really see a reason why this should include the git logic at all.

File: Makefile (hunk 2 above)
Comment (flouthoc) [reply Makefile:63]: `cargo` does treat publishing from another repo as a clean publish, you will have to force `--allow-dirty` which will still work but its not the correct way.

File: Makefile (hunk 3 above)
Comment (flouthoc) [reply Makefile:71]: done.

**Comment by flouthoc** [comment]
@Luap99 Cargo treats publishing from any other repo as a `dirty` release it expects that every release has to be done from the original repo and from the `tag` anything else is considered as a `dirty` release, `dirty` release is still allowed `--allow-dirty` but it is not the right practice which we should follow.

The whole git logic is to prevent dirty release overall.

File: Makefile (hunk 2 above)
Comment (flouthoc) [reply Makefile]: @Luap99 let me try again if `cargo` warns when publishing from a fork or not.

Comment (flouthoc) [reply Makefile]: removed the check i think fork is also allowed

**Comment by Luap99** [comment]
> @Luap99 Cargo treats publishing from any other repo as a `dirty` release it expects that every release has to be done from the original repo and from the `tag` anything else is considered as a `dirty` release, `dirty` release is still allowed using forced `--allow-dirty` but it is not the right practice which we should follow.
> 
> The whole git logic is to prevent dirty release overall.

But I have to original repo locally. Git is distributed all the changes are in my local repo unless I decided to only pull certain changes, all you have is different remotes configured but I don't see why cargo would care about that.

The cargo docs speak about a dirty repo, it git terms that mean uncommitted files. `git diff --quiet`. That should simply not be the case when you make a release you really do not need to pull the whole repo again for that.

I would also not include the git witch logic in the makefile, the author should normally already be on the correct commit.

If you want to make sure we only publish from correct git tags you could check `git describe --tags` matches the version Cargo.toml.

**Comment by flouthoc** [comment]
@Luap99 @baude @mheon PTAL

**Review by Luap99 — COMMENTED** [review]
LGTM

**Comment by flouthoc** [comment]
@baude PTAL
